mssql-plugin/src/e2e-test/java/io.cdap.plugin/mssql/BQValidation.java

```diff
@@ -0,0 +1,238 @@
+     * @throws ParseException If an error occurs while parsing the data.
+     */
+
+    public static boolean compareResultSetAndJsonData(ResultSet rsSource, List<JsonObject> bigQueryData)
+            throws SQLException, ParseException {
+        ResultSetMetaData mdSource = rsSource.getMetaData();
+        boolean result = false;
+        int columnCountSource = mdSource.getColumnCount();
+
+        if (bigQueryData == null) {
+            Assert.fail("bigQueryData is null");
+            return result;
+        }
+
+        // Get the column count of the first JsonObject in bigQueryData
+        int columnCountTarget = 0;
+        if (bigQueryData.size() > 0) {
+            columnCountTarget = bigQueryData.get(0).entrySet().size();
+        }
+        // Compare the number of columns in the source and target
+        Assert.assertEquals("Number of columns in source and target are not equal",
+                columnCountSource, columnCountTarget);
+
+        //Variable 'jsonObjectIdx' to track the index of the current JsonObject in the bigQueryData list,
+        int jsonObjectIdx = 0;
+        while (rsSource.next()) {
+            int currentColumnCount = 1;
+            while (currentColumnCount <= columnCountSource) {
+                String columnTypeName = mdSource.getColumnTypeName(currentColumnCount);
+                int columnType = mdSource.getColumnType(currentColumnCount);
+                String columnName = mdSource.getColumnName(currentColumnCount);
+
+                switch (columnType) {
+
+                    case Types.BIT:
+                        boolean source = rsSource.getBoolean(currentColumnCount);
+                        boolean target = bigQueryData.get(jsonObjectIdx).get(columnName).getAsBoolean();
+                        Assert.assertEquals("Different values found for column : %s", source, target);
+                        break;
+
+                    case Types.DECIMAL:
+                    case Types.NUMERIC:
+                        Decimal sourceDecimal = org.apache.spark.sql.types.Decimal.fromDecimal(rsSource.getBigDecimal(currentColumnCount));
+                        Decimal targetDecimal = Decimal.fromDecimal(bigQueryData.get(jsonObjectIdx).get(columnName).getAsBigDecimal());
+                        Assert.assertEquals("Different values found for column : %s", sourceDecimal, targetDecimal);
+                        break;
+
+                    case Types.REAL:
+                        float sourceReal = rsSource.getFloat(currentColumnCount);
+                        float targetReal = bigQueryData.get(jsonObjectIdx).get(columnName).getAsFloat();
+                        Assert.assertTrue(String.format("Different values found for column : %s", columnName),
+                                Double.compare(sourceReal, targetReal) == 0);
+                        break;
+
+                    case Types.DOUBLE:
+                    case Types.FLOAT:
```

Comment:
Cases for Double and Float shouldn't be separate ?

Reply:
Done